packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -1060,4 +1064,108 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
   protected _addLimitAndOffset(_options: AddLimitOffsetOptions): string {
     throw new Error(`_addLimitAndOffset has not been implemented in ${this.dialect.name}.`);
   }
+
+  /**
+   * Returns a query that creates a savepoint.
+   *
+   * @param savepointName
+   */
+  createSavepointQuery(savepointName: string): string {
+    if (!this.dialect.supports.savepoints) {
+      throw new Error(`Savepoints are not supported by ${this.dialect.name} dialect`);
+    }
+
+    return `SAVEPOINT ${this.quoteIdentifier(savepointName)}`;
+  }
+
+  /**
+   * Returns a query that starts a transaction.
+   *
+   * @param options
+   */
+  startTransactionQuery(options?: StartTransactionQueryOptions): string {
+    if (this.dialect.supports.connectionTransactionMethods) {
+      throw new Error(`startTransactionQuery is not supported by ${this.dialect.name} dialect`);
+    }
+
+    const START_TRANSACTION_QUERY_SUPPORTED_OPTIONS = new Set<keyof StartTransactionQueryOptions>();
+
+    if (this.dialect.supports.startTransaction.readOnly) {
+      START_TRANSACTION_QUERY_SUPPORTED_OPTIONS.add('readOnly');
+    }
+
+    if (options) {
+      rejectInvalidOptions(
+        'startTransactionQuery',
+        this.dialect.name,
+        START_TRANSACTION_QUERY_SUPPORTABLE_OPTIONS,
+        START_TRANSACTION_QUERY_SUPPORTED_OPTIONS,
+        options,
+      );
+    }
+
+    return joinSQLFragments([
+      this.dialect.supports.startTransaction.useBegin ? 'BEGIN' : 'START',
+      this.dialect.supports.startTransaction.transactionType
+        ? options?.transactionType ?? this.sequelize.options.transactionType
```

Comment:
shouldn't `transactionType` be one of the `START_TRANSACTION_QUERY_SUPPORTABLE_OPTIONS`?

Reply:
I've updated this now but had to make a few changes as this was being set by the `normalizeTransactionOptions` function.

I moved the logic to the SQLite query-generator since this is the only dialect that supports it.